File: packages/js-sdk/src/regex/detectors/cl/rut.ts
// Chilean RUT (Rol Unico Tributario) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { clRutChecksum } from '../../validators'

class ClRutDetector extends RegexDetector {
  entityType = EntityType.CL_RUT
  score = 0.85
  contextKeywords = ['rut', 'run', 'rol unico tributario', 'rol unico nacional']
  contextRequired = true

  // XX.XXX.XXX-X format (check digit can be 0-9 or K)
  pattern = /\b\d{1,2}\.?\d{3}\.?\d{3}-?[\dkK]\b/g
  validator = clRutChecksum
}

registerRegion('cl', ClRutDetector)

export { ClRutDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/cl/rut.py
"""Chilean RUT (Rol Unico Tributario) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import cl_rut_checksum


@register_region("cl")
class ClRutDetector(RegexDetector):
    entity_type = EntityType.CL_RUT
    score = 0.85
    context_keywords = [
        "rut", "run", "rol unico tributario", "rol unico nacional",
    ]
    context_required = True
    context_window = 50

    # XX.XXX.XXX-X format (check digit can be 0-9 or K)
    pattern = re.compile(
        r"\b\d{1,2}\.?\d{3}\.?\d{3}-?[\dkK]\b"
    )
    validator = staticmethod(cl_rut_checksum)
